Feature/implement_evaluation_controller_be

VladyslavCherednychenko opened the pull request:
make ef migration to add some new data just for test 
add evaluation controller

VladyslavCherednychenko committed:
feature evaluation controller

make ef migration to add some new data just for test
add evaluation controller

DmytroDorodnykhTeamInternational commented on Portal_Perfomance_Employes/Controllers/UserEvaluationController.cs:
```diff
@@ -0,0 +1,139 @@
+            var empSkills = await _context.EmployeeSkills.Where(e => e.EmployeeId == id).ToListAsync();
+            if (empSkills != null)
+            {
+                var hardSkillsTable = new DataTable();
```
can be moved to a func, same code

VladyslavCherednychenko replied:
will be done

DmytroDorodnykhTeamInternational commented on Portal_Perfomance_Employes/Controllers/UserEvaluationController.cs:
```diff
@@ -0,0 +1,139 @@
+        [HttpPut]
+        public async Task<IActionResult> AddUserSkill(int skillType, string skillName, int skillLevelId, int skillLevelTypeId)
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);
```
if fails to parse and 0 by default? could cause some issues if we have something with id 0 in db

VladyslavCherednychenko replied:
will be done! Also nooticed that I made same mistake in other controller so I'll fix them also

dangoth commented on Portal_Perfomance_Employes/Data/AppDbContext.cs:
```diff
@@ -230,6 +230,26 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 new TeamMember { Id = 10, EmployeeId = 11, TeamId = 2, FirstName = "Emp11", LastName = "Test11", FromDate = DateTime.Now, IsActive = true },
                 new TeamMember { Id = 11, EmployeeId = 4, TeamId = 2, FirstName = "Emp4", LastName = "Test4", FromDate = DateTime.Now, IsActive = true }
                 );
+            modelBuilder.Entity<EmployeeSkill>().HasData(
```
Can you use data seeding from my last PR so we don't have double population of the same IDs or conflicts when merging? Alternatively I can overwrite mine

VladyslavCherednychenko replied:
Thanks for the notice, I'll do it in a new commit

VladyslavCherednychenko committed:
rebase function of user skills request by moving similar functionality to another function
add entity seeding from Maciek pr so now we shouldn't have a conflict
fix vulnerability of TryParse method in users controller that possibly can prevent to unwanted access to db
fix an error: “An exception has been raised that is likely due to a transient failure. Consider enabling transient error resiliency by adding 'EnableRetryOnFailure' to the 'UseSqlServer' call.” by adding new instruction to program.cs
